SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialActorChannel.cpp

```diff
@@ -739,9 +743,32 @@ int64 USpatialActorChannel::ReplicateActor()
 	if (NetDriver->LoadBalanceStrategy != nullptr &&
 		NetDriver->StaticComponentView->HasAuthority(EntityId, SpatialConstants::AUTHORITY_INTENT_COMPONENT_ID))
 	{
-		if (!NetDriver->LoadBalanceStrategy->ShouldHaveAuthority(*Actor) && !NetDriver->LockingPolicy->IsLocked(Actor))
+		if (MigrationVirtualWorkerId != SpatialConstants::INVALID_VIRTUAL_WORKER_ID)
+		{
+			if(MigrationFrame != NetDriver->ReplicationFrame)
+			{
+				Sender->SendAuthorityIntentUpdate(*Actor, MigrationVirtualWorkerId);
+
+				// If we're setting a different authority intent, preemptively changed to ROLE_SimulatedProxy
+				Actor->Role = ROLE_SimulatedProxy;
+				Actor->RemoteRole = ROLE_Authority;
+
+				Actor->OnAuthorityLost();
+				MigrationVirtualWorkerId = SpatialConstants::INVALID_VIRTUAL_WORKER_ID;
+			}
+		}
+		else if (!NetDriver->LoadBalanceStrategy->ShouldHaveAuthority(*Actor) && !NetDriver->LockingPolicy->IsLocked(Actor))
 		{
-			const AActor* NetOwner = Actor->GetNetOwner();
+			AActor* NetOwner = [this]
+			{
+				AActor* HierarchyRoot = SpatialGDK::GetHierarchyRoot(Actor);
+				if(HierarchyRoot)
+				{
+					return HierarchyRoot;
+				}
+				return Actor;
+			}();
+			
 
 			uint64 HierarchyAuthorityReceivedTimestamp = AuthorityReceivedTimestamp;
 			if (NetOwner != nullptr)
```

Comment:
This check doesn't make sense anymore since the lambda returns the Actor itself if the root doesn't exist